include/matx/executors/stf.h

```diff
@@ -0,0 +1,253 @@
+////////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (c) 2021, NVIDIA Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "matx/core/defines.h"
+#include "matx/executors/host.h"
+#include "matx/executors/kernel.h"
+
+#include <cuda/experimental/stf.cuh>
+
+//using namespace cuda::experimental::stf;
+//using namespace cudastf;
+
+namespace matx
+{
+
+
+/* Albert - Needed to declare this here to avoid compile error. */
+template <typename T> constexpr bool is_matx_op_lvalue();
+template <typename T> constexpr bool is_matx_set_op();
+
+  class stfExecutor {
+    public:
+      using matx_cuda = bool;  // signal this is a GPU executor
+      using matx_executor = bool; ///< Type trait indicating this is an executor
+//using cudastf::async_resources_handle;
+//using cudastf::stream_ctx;
+      /**
+       * @brief Construct a new stfExecutor with a stream
+       * 
+       * @param stream CUDA stream
+       */
+      stfExecutor(cudaStream_t stream) : stream_(stream) {
```

Comment:
What does a stream do here? I thought STF had its own internal streams?

Reply:
@cliffburdick In STF you can create nested/localized contexts & streams from existing (non-STF created) streams. This allows STF mechanisms to be correctly synchronized within the existing stream ecosystem. @caugonnet correct me if I am wrong.